File: src/ast/NameType.java
package ast;

import java.io.StringWriter;

public class NameType extends BaseNode {
    private String _nameValue;

    public NameType(String nameValue, NodeType type) {
        super(type);
        _nameValue = nameValue;
    }

    public NameType(String nameValue) {
        super(NodeType.NameType);
        _nameValue = nameValue;
    }

    @Override
    public String GetName() {
        return _nameValue;
    }

    @Override
    public void PrintLeft(StringWriter writer) {
        writer.write(_nameValue);
    }
}

File: src/de/fearlesstobi/demangler/ast/NameType.kt
package de.fearlesstobi.demangler.ast

import java.io.StringWriter

class NameType : BaseNode {
    private val nameValue: String

    constructor(nameValue: String, type: NodeType) : super(type) {
        this.nameValue = nameValue
    }

    constructor(nameValue: String) : super(NodeType.NameType) {
        this.nameValue = nameValue
    }

    override fun getName(): String? {
        return nameValue
    }

    public override fun printLeft(writer: StringWriter) {
        writer.write(nameValue)
    }
}